src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateMethod/GenerateMethodTests.vb

```diff
@@ -4191,6 +4191,183 @@ Class Program
 End Class")
         End Function
 
+                    <WorkItem(16975, "https://github.com/dotnet/roslyn/issues/16975")>
+            <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateMethod)>
+            Public Async Function TestWithSameMethodNameAsTypeName1() As Task
+                Await TestInRegularAndScriptAsync(
+"Imports System
+Class C
+    Sub Bar()
+        [|Foo|]()
```

Comment:
Note: for VB we *probably* do not want to offer generate-method if you've just written Foo without parens.  In that case, it's just not that likely htat you were attempting to create a method with that name, but were instead trying to reference the type.

If that's hte current behavior, great.  Just please add a test.  If that isn't the behavior, then we should add a check to have that behavior.